matrix_content_scanner/utils/encrypted_file_metadata.py

```diff
@@ -0,0 +1,95 @@
+#  Copyright 2022 The Matrix.org Foundation C.I.C.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from jsonschema import ValidationError, validate
+
+from matrix_content_scanner.utils.constants import ErrCode
+from matrix_content_scanner.utils.errors import ContentScannerRestError
+from matrix_content_scanner.utils.types import JsonDict
+
+# This is a subset of the content of an m.room.message event that includes a file, with
+# only the info that we need to locate and decrypt the file.
+_encrypted_file_metadata_schema = {
+    "type": "object",
+    "required": ["file"],
+    "properties": {
+        "file": {
+            "type": "object",
+            "required": ["v", "iv", "url", "hashes", "key"],
+            "properties": {
+                "v": {"const": "v2"},
+                "iv": {"type": "string"},
+                "url": {"type": "string"},
+                "hashes": {
+                    "type": "object",
+                    "required": ["sha256"],
+                    "properties": {
+                        "sha256": {"type": "string"},
+                    },
+                },
+                "key": {
+                    "type": "object",
+                    "required": ["alg", "kty", "k", "key_ops"],
+                    "properties": {
+                        "alg": {"const": "A256CTR"},
+                        "kty": {"const": "oct"},
+                        "k": {"type": "string"},
+                        "key_ops": {"type": "array", "items": {"type": "string"}},
```

Comment:
Do we want to assert that this contains `"encrypt"` and `"decrypt"`?

Reply:
Oh I see we do this in Python below!

Reply:
Yeah I couldn't figure out a satisfactory way to validate this with jsonschema.